planner/core/expression_rewriter_test.go

```diff
@@ -58,3 +59,73 @@ func (s *testExpressionRewriterSuite) TestBinaryOpFunction(c *C) {
 	tk.MustQuery("SELECT * FROM t WHERE (a,b,c) <= (1,2,3) order by b").Check(testkit.Rows("1 1 <nil>", "1 2 3"))
 	tk.MustQuery("SELECT * FROM t WHERE (a,b,c) > (1,2,3) order by b").Check(testkit.Rows("1 3 <nil>"))
 }
+
+func (s *testExpressionRewriterSuite) TestDefaultFunction(c *C) {
+	defer testleak.AfterTest(c)()
+	store, dom, err := newStoreWithBootstrap()
+	c.Assert(err, IsNil)
+	tk := testkit.NewTestKit(c, store)
+	defer func() {
+		dom.Close()
+		store.Close()
+	}()
+	tk.MustExec("use test")
+	tk.MustExec("drop table if exists t1")
+	tk.MustExec(`create table t1(
+		a varchar(10) default 'def',
+		b varchar(10),
+		c int default '10',
+		d double default '3.14',
+		e datetime default '20180101',
+		f datetime default current_timestamp);`)
+	tk.MustExec("insert into t1(a, b, c, d) values ('1', '1', 1, 1)")
+	tk.MustQuery(`select
+		default(a) as defa,
+		default(b) as defb,
+		default(c) as defc,
+		default(d) as defd,
+		default(e) as defe,
+		default(f) as deff
+		from t1`).Check(testutil.RowsWithSep("|", "def|<nil>|10|3.14|2018-01-01 00:00:00|<nil>"))
+	err = tk.ExecToErr("select default(x) from t1")
+	c.Assert(err.Error(), Equals, "[planner:1054]Unknown column 'x' in 'field list'")
+
+	tk.MustQuery("select default(a0) from (select a as a0 from t1) as t0").Check(testkit.Rows("def"))
+	err = tk.ExecToErr("select default(a0) from (select a+1 as a0 from t1) as t0")
+	c.Assert(err.Error(), Equals, "[table:1364]Field 'a0' doesn't have a default value")
+
+	tk.MustExec("create table t2(a varchar(10), b varchar(10))")
+	tk.MustExec("insert into t2 values ('1', '1')")
+	err = tk.ExecToErr("select default(a) from t1, t2")
+	c.Assert(err, NotNil)
```

Comment:
it's better to check the error msg as well.

Reply:
addressed, thanks!